spec/controllers/api/v1/users_controller_spec.rb

```diff
@@ -199,6 +199,96 @@
         end
       end
     end
+
+    describe 'GET /mentionable_users' do
+      let!(:reviewer) { create(:user, :reviewer) }
+      let!(:donor) { create(:user)}
+      let!(:supervisor) { create(:user, :supervisor) }
+      let!(:order_administrator) { create(:user, :order_administrator) }
+      let!(:charity) { create(:user, :charity) }
+      let!(:order_fulfilment) { create(:user, :order_fulfilment) }
+      let!(:offer) { create(:offer, reviewed_by: reviewer, created_by: user) }
+      let!(:order) { create(:order, created_by: charity) }
+      before { generate_and_set_token(user) }
+
+      it 'returns 200' do
+        get :mentionable_users, app_name: ADMIN_APP, offer_id: offer.id, is_private: false
+        expect(response).to have_http_status(:success)
+      end
+
+      context 'if no messageable id is passed in params' do
+        it 'return empty array' do
+          get :mentionable_users, app_name: ADMIN_APP, offer_id: nil, is_private: false
+          expect(parsed_body['users']).to be_empty
+        end
+      end
+
+      context 'if public' do
+        context 'admin app' do
+          it 'returns reviewers and donors' do
+            generate_and_set_token(supervisor)
+            get :mentionable_users, app_name: ADMIN_APP, offer_id: offer.id, is_private: false
+            users = [[User.supervisors.map(&:id), User.reviewers.map(&:id)].flatten - [supervisor.id]].flatten.map { |id| {'id' => id, 'name' => User.find(id).full_name } }
+            expect(parsed_body['users']).to match_array(users)
+          end
+
+          it 'returns donors and supervisors' do
+            generate_and_set_token(reviewer)
+            get :mentionable_users, app_name: ADMIN_APP, offer_id: offer.id, is_private: false
+            users = [[User.supervisors.map(&:id), User.reviewers.map(&:id)].flatten - [reviewer.id]].flatten.map { |id| {'id' => id, 'name' => User.find(id).full_name} }
+            expect(parsed_body['users']).to match_array(users)
+          end
+        end
+
+        context 'donor app' do
+          it 'returns reviewers and supervisors' do
```

Comment:
how about a spec for `is_private: true` to confirm donor's can't access anything extra just by changing a param